Radiative Equilibrium tutorial (#1979)
CliMA/ClimaAtmos.jl: docs/src/radiative_equilibrium.md

> # Example

Author: Sounds good, thank you so much!

A suite of concrete examples are provided here as a guidance for running single column Radiative Equilibrium.  

# Radiative Equilibrium

> Radiative Equilibrium can be a useful tool for analyzing atmospheric temperature profiles with differing conditions. To run a radiative equilibrium simulation create a yaml file (or use a default one) and run the following commands in your terminal (after instantiating the necessary packages):

Reviewer: ```suggestion
Radiative equilibrium is a useful framework for analyzing atmospheric temperature profiles with different atmosphere conditions. In radiative equilibrium, we only consider heat exchange from radiation.

To run a radiative equilibrium simulation, create a yaml file (or use a default one) and run the following commands in your terminal (after instantiating the necessary packages):
```

The yaml file should look something like this:
`mysim.yml`
```
rad: "clearsky" 
idealized_h2o: true 
hyperdiff: "false" 
config: "column" 
initial_condition: "IsothermalProfile" 
z_max: 70000
z_elem: 70 
dz_bottom: 100 
dz_top: 10000 
t_end: "654days" 
dt: "3hours" 
dt_save_to_sol: "30hours" 
dt_save_to_disk: "100days" 
prognostic_surface: "true" 
job_id: "single_column_radiative_equilibrium_clearsky_prognostic_surface_temp"
```

> Then the command line input should look like this

Reviewer: ```suggestion
Then run the simulation with the following command:
```

```
julia --project=examples examples/driver.jl --config_file path to mysim.yml
```

> The yaml inputs above can be changed to adjust the kind of simulation desired. Options such as t_end can just be increased to the desired length, other arguments such as radiation type can be changed based on the options in `default_config 2.yml` and `default_config.yml`. Radiative equilibrium is ran with clearsky radiation, a prognostic surface, and generally a single column model.

Reviewer: ```suggestion
The configuration in the yaml file above can be changed to adjust to the kind of simulation desired. Options such as t\_end can be increased to the desired simulation time. Other arguments such as radiation type can be changed based on the options in `default_config 2.yml` and `default_config.yml`.
```

Reviewer: (You can see how the docs are rendered by clicking on deployment at the bottom of your PR. What is the default_config 2.yml?)

Author: I think default_config 2.yml is like the equivalent of cli_options.jl since the new update? cli_options.jl was taken out so I'm not sure where to point to for the possible simulation arguments?

Reviewer: I think all the possible arguments that matter here is in `default_config.yml`

```
julia --project=examples examples/driver.jl --config_file path to mysim.yml
```

The configuration in the yaml file above can be changed to adjust to the kind of simulation desired. Options such as t\_end can be increased to the desired simulation time. Other arguments such as radiation type can be changed based on the options in `default_config 2.yml` and `default_config.yml`.

After a simulation is ran the output files should be available in the output folder, where hdf5 files store all the simulation's data, and two mp4s should be produced with a vertical temperature profile and a vertical wind profile which change over time. If you want to quickly access data from the final day of the simulation simply add a -i to the command line input, like so:

```
julia -i --project=examples examples/driver.jl --config_file filepath
```

This will bring you into a julia environment where the current state of the simulation can be accessed. Run propertynames(Y) and propertynames(p) to see accesible values, some examples:

![](assets/sample_-i_outputs.png)

Here is a typical vertical temperature profile on the final day of a simulation (the dot is surface temperature):

> ![](assets/example_temp_profile.png)

Reviewer: A minor issue. But it would be better if you could change the y-axis from elevation to height, to be consistent with other plots.

Note: The surface temperature must also converge before the simulation reaches radiative equilibrium, here is a typical convergence of surface temperature plot:

![](assets/surface_temp_convergence.png)

# Greenhouse Gas Concentrations
In `radiation.jl` one can change the concentrations of greenhouse gases in order to test the affects on the atmospheric temperature profile.

> To change the ozone concentration multiply the following variable the desired coefficent (line 89):

Reviewer: Instead of adding a screenshot here, I think it would be better to link to the actual code, as the line number could change.

![](assets/O3_sample_code.png)

> Then comparing the results between ozone simulations could look something like this:

Reviewer: ```suggestion
Here are the equilibrium temperature profiles under different ozone concentrations:
```

![](assets/O3_temp_profile.png)
![](assets/O3_temp_difference.png)

> To change the CO2 concentration multiply the volume mixing ratio by the desired coefficient (line 100):

Reviewer: ```suggestion
To change the CO2 concentration, multiply the volume mixing ratio by the desired coefficient (line 100):
```
Same here, it would be better to link to the code instead of having a screen shot.

![](assets/CO2_sample_code.png)

> Results from increasing CO2 concentration could look something like this:

Reviewer: ```suggestion
Here are the equilibrium temperature profiles under different ozone concentrations:
```

![](assets/CO2_temp_profile.png)
![](assets/CO2_temp_difference.png)

> A similar process to O3 and CO2 can work for increasing the concentration of any other gases in the atmosphere

Reviewer: ```suggestion
Similarly, one can change the concentration of any other gases in the atmosphere
```

# Data extraction in Julia
To access the data from hdf5 files one can use the following code to create a var `diagnostics` which stores values from a hdf5 file path:

```
import ClimaAtmos: time_from_filename
import ClimaCore: Geometry, Spaces, Fields, InputOutput
import ClimaComms
import CairoMakie: Makie
import Statistics: mean

function read_hdf5_file(file_path)
    reader =
        InputOutput.HDF5Reader(file_path, ClimaComms.SingletonCommsContext())
    diagnostics = InputOutput.read_field(reader, "diagnostics")
    close(reader)
    return time_from_filename(file_path), diagnostics
end

time, diagnostics = read_hdf5_file("sample/file/path")
```

> `diagnostics` will contain the values from the state of the simulation on a given day. propertynames(diagnostics) can be useful for finding accessible values. Example: `diagnostics.sfc_temperature` will result in the surface temperature at that day in the simulation.

Reviewer: ```suggestion
`diagnostics` will contain the values from the state of the simulation at a given time. `propertynames(diagnostics)` can be useful for finding accessible values. For example, `diagnostics.sfc_temperature` contains the surface temperature at a given time in the simulation.
```
